scaling_performance/cluster_limits.adoc

```diff
@@ -0,0 +1,102 @@
+[[scaling-performance-cluster-limits]]
+= Cluster Limits
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+This topic summarizes the limits for objects in {product-title}.
+
+In most cases, exceeding these thresholds results in lower overall performance.
+It does not necessarily mean that the cluster will fail over.
+
+Some of the limits represented in this topic are given for the largest possible
+cluster. For smaller clusters, the limits are proportionally lower.
+
+There are many factors that influence the stated thresholds, including the etcd
+version or storage data format.
+
+[[scaling-performance-current-cluster-limits]]
+== {product-title} 3.7 Cluster Limits
+
+[options="header",cols="3,2"]
+|===
+| Limit Type |Current Limit
+
+|Total number of all objects footnoteref:[allobjects,There is a limitation on the total number of objects on the system, as this affects etcd and its resource consumption.]
+| 250,000
+
+| Number of nodes footnoteref:[numberofnodes,Having clusters with more than 5000 nodes is not the best option. Consider splitting into multiple clusters.]
+| 2,000
+
+| Number of pods
+| 120,000
+
+| Number of xref:../admin_guide/manage_nodes.adoc#admin-guide-max-pods-per-node[pods per node]
+| 250
+
+| Number of xref:../admin_guide/manage_nodes.adoc#admin-guide-max-pods-per-node[pods per core]
+| 10 is the default value. The maximum supported value is the number of pods per node.
+
+| Number of namespaces
+| 10,000
+
+| Number of pods per namespace footnoteref:[objectpernamespace,There are
+a number of control loops in the system that need to iterate over all objects
+in a given namespace as a reaction to some changes in state. Having a large
+number of objects of a given type in a single namespace can make those loops
+expensive and slow down processing given state changes.]
+| 15,000
+
+| Number of services footnoteref:[servicesandendpoints,Each service port and each service back-end has a corresponding entry in iptables. The number of back-ends of a given service impact the size of the endpoints objects, which impacts the size of data that is being sent all over the system.]
+| 10,000
+
+| Number of back-ends per service
+| 5,000
+
+| Number of deployments per namespace footnoteref:[objectpernamespace]
+| 20,000
+
+|===
+
+[[scaling-performance-planning-your-environment-according-to-cluster-limits]]
+== Planning Your Environment According to Cluster Limits
+
+[IMPORTANT]
+====
+Oversubscribing the physical resources on a node affects resource guarantees the
+Kubernetes scheduler makes during pod placement. Learn what measures you can
+take to xref:../../admin_guide/overcommit.adoc#disabling-swap-memory[avoid memory swapping].
+====
+
+While
+xref:../install_config/install/planning.adoc#install-config-install-planning[planning
+your environment], determine how many pods are expected to fit per node:
```

Comment:
@jeremyeder Should we add a short discussion around how the number of pods expected to fit on a node is dependent on the application itself?  i.e. there is no magic in a pod - you are still responsible for understanding the memory/cpu/storage requirements of the application.   Maybe outside the scope of this section, but still a key part of the "pods per node" thing.